device/celestica/x86_64-cel_silverstone-r0/plugins/sfputil.py

```diff
@@ -477,7 +496,15 @@ def _init_cmis_module(self, port_num):
         #log_info("_init_cmis_module: p={0}, id='{1}', name='{2}', part='{3}'".format(port_num, buf[0], name, part))
         xcvr = name.upper() + " " + part.upper()
 
-        return self._init_cmis_module_custom(port_num, xcvr, self.hwsku)
+        # Dispatch as per the CMIS revision id
+        buf = self._read_eeprom_devid(port_num, self.IDENTITY_EEPROM_ADDR, 0x0, 4)
+        rev = 0x30
+        if buf is not None:
+            rev = int(buf[1], 16)
+        if rev >= 0x40:
+            return self._init_cmis4_module(port_num, xcvr, self.hwsku)
+
```

Comment:
All of the CMIS3 modules we have are working fine with 4x100G in its default application mode, and hence this init sequence will only be performed on CMIS4 modules.

Note: The definition of the datapath initialiation slightly differs from CMIS3 to CMIS4, and hence we can't use _init_cmis4_module() on CMIS3 modules.